### PR TITLE
APL-2322 Move Amazon linux install script test to new-e2e

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Move Amazon Linux tests to new-e2e

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/tests/agent-platform/common/test_client.go [hunk 1]
```diff
@@ -58,6 +58,10 @@ func getPackageManager(vmClient e2eClient.VM) PackageManager {
 	if _, err := vmClient.ExecuteWithError("command -v apt"); err == nil {
 		return pkgmanager.NewAptPackageManager(vmClient)
 	}
+
+	if _, err := vmClient.ExecuteWithError("command -v yum"); err == nil {
+		return pkgmanager.NewYumPackageManager(vmClient)
+	}
```
Comment (spencergilbert) [inline test/new-e2e/tests/agent-platform/common/test_client.go:64]: **💭 thought:** ‏Probably a very unlikely edge case we shouldn't worry about, but it's possible to have `yum` installed without it being used as the OS package manager.

**Review by Pythyu — APPROVED** [review]
LGTM if CI is green

**Review by spencergilbert — APPROVED** [review]

File: test/new-e2e/tests/agent-platform/common/test_client.go (hunk 1 above)
Comment (KevinFairise2) [reply test/new-e2e/tests/agent-platform/common/test_client.go:64]: Ahaha that would be very silly 🤣 . I think it should not an issue since we chose the AMI we use in the test. And as long as we don't find a very strange AMI, that should never happen

Comment (spencergilbert) [reply test/new-e2e/tests/agent-platform/common/test_client.go:64]: Yep, highly unlikely we hit it - but figured it was possible given how we were checking 😉
